xbmc/pvr/windows/GUIWindowPVRBase.cpp

```diff
@@ -105,9 +105,19 @@ bool CGUIWindowPVRBase::OnAction(const CAction &action)
   {
     case ACTION_PREVIOUS_CHANNELGROUP:
     case ACTION_NEXT_CHANNELGROUP:
+    {
       // switch to next or previous group
       SetChannelGroup(action.GetID() == ACTION_NEXT_CHANNELGROUP ? m_channelGroup->GetNextGroup() : m_channelGroup->GetPreviousGroup());
       return true;
+    }
+    case ACTION_MOVE_RIGHT:
+    case ACTION_MOVE_LEFT:
+    case ACTION_MOVE_UP:
+    case ACTION_MOVE_DOWN:
+    {
+      if (IsChannelGroupsControlFocused())
+        return CGUIMediaWindow::OnAction(action) && ActivateSelectedChannelGroup();
```

Comment:
what's the reason for calling `CGUIMediaWindow::OnAction(action)` first?

Reply:
this is need to first update the control and later switch to the group, right?

Reply:
Yes.